### DISCUSSION TITLE
Detached object unexpected behavior

### DISCUSSION THREAD

**Discussion by Arbiter-S** [sqlalchemy/sqlalchemy#11008] [Usage Questions] [answered]
I'm experiencing unexpected behavior about detached object.

```
engine = create_engine("sqlite:///database.db")

with Session(engine) as session:
    price = session.get(Price, 1)


print(price in session)
print(price.price_per_unit)
```

output:
```
False
130000
```

Following this tutorial https://docs.sqlalchemy.org/en/20/tutorial/orm_data_manipulation.html I expected the objects attributes to not be loaded and therefore get a DetachedInstanceError.

**Comment by zzzeek** [comment, marked as answer]
local column attributes are not automatically expired when an object becomes detached.   expiration events include calling `session.expire_all()`, or `session.commit()` if the `expire_on_commit` parameter is left at its default value of `True`.

**Reply by Arbiter-S** [reply]
Appreciate it. The document uses session.close() which does not behave as it was shown.

**Reply by zzzeek** [reply]
.close() does not expire objects, it only detaches them.   if you can show where in the document you think it's incorrect (that doc runs via doctest so it's tested code) we can fix it

**Reply by CaselIT** [reply]
I think there is not overly clear https://docs.sqlalchemy.org/en/20/tutorial/orm_data_manipulation.html#closing-a-session in particular the
> It expunges all objects from the [Session](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session).

then shows a detached error

**Reply by zzzeek** [reply]
that's because that attribute is expired.  so it's not incorrect, just misleading, and if someone misses the line in the paragraph "In particular, we will note that objects that were still in an [expired](https://docs.sqlalchemy.org/en/20/glossary.html#term-expired) state, for example due to the call to [Session.commit()](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.commit)," and just looks at the code they can be confused.   lets add a comment above it saying "note that squidward.name was just expired previously, so its value is unloaded"

**Reply by CaselIT** [reply]
> lets add a comment above it saying "note that squidward.name was just expired previously, so its value is unloaded"

are you doing that?

**Reply by zzzeek** [reply]
im working for the boss today, sorry

**Reply by CaselIT** [reply]
will send a patch

**Reply by CaselIT** [reply]
added a comment in the example here 1e099d1a855e492389c02559d2059d93e5a5a091
